src/components/popover/popover.jsx

```diff
@@ -53,46 +53,55 @@ const Popover = forwardRef(function Popover(props, ref) {
 
   return (
     <section
-      aria-labelledby="popover-heading"
-      className={styles.container}
+      aria-labelledby={props.hideHeading ? undefined : 'popover-heading'}
```

Comment:
If we're hiding the heading, we will want an `aria-label` for accessibility. i.e. what is the fly out menu's purpose? e.g. "Settings for Notifications page", "Actions on resource Foo row".